source/loop.cpp

```diff
@@ -420,12 +546,60 @@ void Loop::CalculateAbundanceCorrection(double helium_to_hydrogen_ratio)
   parameters.ion_mass_correction = (1.0 + 4.0*helium_to_hydrogen_ratio)/(2.0 + 3.0*helium_to_hydrogen_ratio)*(1.0 + z_avg)/z_avg;
 }
 
+void Loop::ReadRadiativeLossData()
+{
+    // Reads in the radiative loss files.  Only need to do once during the setup.
+   std::string path = "data/radiation/";
+   std::string filenames[] = {"abund_10_rad_loss.dat","abund_15_rad_loss.dat","abund_20_rad_loss.dat","abund_25_rad_loss.dat",
+                                "abund_30_rad_loss.dat","abund_35_rad_loss.dat","abund_40_rad_loss.dat"};
+   std::ifstream fin;
+   int n_files = sizeof(filenames)/sizeof(filenames[0]);
```

Comment:
Is this really the only way to get the length a list/array in modern C++?

Reply:
This whole bit of code can be done more elegantly, but requires C++ 17 or more recent.  We don't need to hardcode the filenames, and can instead read in the directory and determine the filenames and number of files.  

Any objection to requiring C++ 17?  I've pushed a commit to do so, but can revert if we'd prefer not.  There does appear to be `boost` functions that could be used in its place.

Reply:
No objection from me!